ros/src/computing/planning/common/lib/openplanner/op_planner/src/BehaviorStateMachine.cpp

```diff
@@ -305,4 +338,123 @@ BehaviorStateMachine* GoalState::GetNextState()
 		return FindBehaviorState(this->m_Behavior); // return and reset
 }
 
+BehaviorStateMachine* MissionAccomplishedState::GetNextState()
+{
+	return FindBehaviorState(this->m_Behavior); // return and reset
+}
+
+BehaviorStateMachine* ForwardStateII::GetNextState()
+{
+	PreCalculatedConditions* pCParams = GetCalcParams();
+
+	if(pCParams->currentGoalID != pCParams->prevGoalID)
+		return FindBehaviorState(GOAL_STATE);
+
+	else if(m_pParams->enableStopSignBehavior
+			&& pCParams->currentStopSignID > 0
+			&& pCParams->currentStopSignID != pCParams->prevStopSignID)
+		return FindBehaviorState(STOP_SIGN_STOP_STATE);
+
+	else if(m_pParams->enableFollowing && pCParams->bFullyBlock)
+		return FindBehaviorState(FOLLOW_STATE);
+
+	else if(m_pParams->enableSwerving
+			&& pCParams->distanceToNext <= m_pParams->minDistanceToAvoid
+			&& !pCParams->bFullyBlock
+			&& pCParams->iCurrSafeTrajectory != pCParams->iPrevSafeTrajectory)
+		return FindBehaviorState(OBSTACLE_AVOIDANCE_STATE);
+
+	else
+		return FindBehaviorState(this->m_Behavior);
+}
+
+BehaviorStateMachine* FollowStateII::GetNextState()
+{
+	PreCalculatedConditions* pCParams = GetCalcParams();
+
+	if(pCParams->currentGoalID != pCParams->prevGoalID)
+		return FindBehaviorState(GOAL_STATE);
+
+	else if(m_pParams->enableStopSignBehavior
+			&& pCParams->currentStopSignID > 0
+			&& pCParams->currentStopSignID != pCParams->prevStopSignID)
+		return FindBehaviorState(STOP_SIGN_STOP_STATE);
+
+	else if(m_pParams->enableSwerving
+			&& pCParams->distanceToNext <= m_pParams->minDistanceToAvoid
+			&& !pCParams->bFullyBlock
+			&& pCParams->iCurrSafeTrajectory != pCParams->iPrevSafeTrajectory)
+		return FindBehaviorState(OBSTACLE_AVOIDANCE_STATE);
+
+	else if(!pCParams->bFullyBlock)
+		return FindBehaviorState(FORWARD_STATE);
+
+	else
+		return FindBehaviorState(this->m_Behavior); // return and reset
+}
+
+BehaviorStateMachine* SwerveStateII::GetNextState()
+{
+	PreCalculatedConditions* pCParams = GetCalcParams();
```

Comment:
sure, but GetCalcParams() never return an invalid point.

Reply:
sure it can. Like if someone has by mistake manipulated the memory address of GetCalcParams. Or if the function was not able allocate the memory because of the mem fragmentation issue. Etc.